src/py/crankshaft/crankshaft/analysis_data_provider.py

```diff
@@ -2,66 +2,57 @@
 import plpy
 import pysal_utils as pu
 
+NULL_VALUE_ERROR = ('No usable data passed to analysis. Check your input rows '
+                    'for null values and fill in appropriately.')
 
-class AnalysisDataProvider:
-    def get_getis(self, w_type, params):
-        """fetch data for getis ord's g"""
+
+def verify_data(f):
+    def wrapper(*args, **kwargs):
         try:
-            query = pu.construct_neighbor_query(w_type, params)
-            result = plpy.execute(query)
-            # if there are no neighbors, exit
-            if len(result) == 0:
-                return pu.empty_zipped_array(4)
+            data = f(*args, **kwargs)
+            if len(data) == 0:
+                plpy.error(NULL_VALUE_ERROR)
             else:
-                return result
-        except plpy.SPIError, err:
-            plpy.error('Analysis failed: %s' % err)
+                return data
+        except Exception, err:
+            plpy.error('Analysis failed: {}'.format(err))
```

Comment:
Good point. I guess returning `[]` would be a good option because the number of values in the tuples is variable depending on the `get_` function run.